### PR TITLE
Optimize GetLongbook GetShortbook queries

### PR DESCRIPTION
**Pull request by Kbhat1**
Adds in optimizations to the `GetLongbook` and `GetShortBook` endpoints
- Dumps the whole shortbook / longbook in one entry of the db for a market instead of separate entries 
- Write process: Queries a shortbook / longbook from rpc node => convert whole shortbook / longbook to string => Compress (gzip) that string => Writes compressed string to table
- Query process: Queries db for entry corresponding to market => Uncompresses string to json bytes => unmarshals into shortbook / longbook struct

The compression is a necessity due to the fact that dynamo limits each attribute to 400KB

### CODE REVIEW

File: app/handler.go [hunk 1]
```diff
@@ -268,10 +314,14 @@ func Catchup() {
...
-		//shortBookResponse := QueryShortbook(TargetContract)
-		//HandleShortBook(shortBookResponse, TargetContract)
+		longBookAllResponses := QueryLongbookAll(TargetContract)
+		HandleLongBookAll(longBookAllResponses, TargetContract)
+		shortBookAllResponses := QueryShortbookAll(TargetContract)
+		HandleShortBookAll(shortBookAllResponses, TargetContract)
+		// longBookResponses := QueryLongbook(TargetContract)
+		// HandleLongBook(longBookResponses, TargetContract)
+		// shortBookResponse := QueryShortbook(TargetContract)
+		// HandleShortBook(shortBookResponse, TargetContract)
```
Comment (ziyincody) [inline app/handler.go:324]: guess we can remove these

File: db/db.go [hunk 2]
```diff
@@ -278,6 +284,86 @@ func WriteLongBook(longBooks []dextypes.LongBook, contractAddr string) {
...
+
+		requests = append(requests, &dynamodb.WriteRequest{
+			PutRequest: &dynamodb.PutRequest{
+				Item: map[string]*dynamodb.AttributeValue{
+					"market": {
+						S: aws.String(fmt.Sprintf("%s-%s-%s", contractAddr, longBook.PriceDenom, longBook.AssetDenom)),
+					},
+					"longbook": {
+						B: b.Bytes(),
+					},
```
Comment (ziyincody) [inline db/db.go:313]: Probably a P1 but I think it makes sense to add `height` as the range key so we have historical long/short book data. But we can also backfill this to a historic data table in the future

File: db/db.go [hunk 3]
```diff
@@ -278,6 +284,86 @@ func WriteLongBook(longBooks []dextypes.LongBook, contractAddr string) {
...
+			panic(fmt.Sprintf("failed to marshal shortbook, %v", err))
+		}
+		var b bytes.Buffer
+		gz := gzip.NewWriter(&b)
+		if _, err := gz.Write(shortBookJSON); err != nil {
+			panic(err)
+		}
+		if err := gz.Close(); err != nil {
+			panic(err)
+		}
```
Comment (ziyincody) [inline db/db.go:343]: nit: create a helper for the compression logic to share

**Review by ziyincody — APPROVED** [review]
LGTM with nit comments, feel free to merge first if you want to unblock ASAP

File: db/db.go (hunk 2 above)
Comment (Kbhat1) [reply db/db.go:313]: Yea this makes sense. I can add this later if we do want historical short / long books

File: db/db.go (hunk 3 above)
Comment (Kbhat1) [reply db/db.go]: Added

**Pull request merged**